### PR TITLE
Fix harvest crawler to skip non-XML files during crawl

### LINKED ISSUE #246: Harvest crawler processes non-XML files unnecessarily
## Checked for duplicates
Yes - I've already checked

## 🐛 Describe the bug
The `PDSProductCrawler` does not filter out non-XML files before attempting to process them during harvest. Files such as CSV inventory files, PDFs, TXT files, and other non-label formats are passed through the crawl pipeline unnecessarily, causing wasted processing and potential errors.

## 🕵️ Expected behavior
The crawler should skip any file that is not an XML file (i.e., does not end with `.xml`), logging it as a SKIP and continuing to the next file.

## 📜 To Reproduce
1. Run `harvest-solr` against a data directory containing non-XML files (e.g., CSV inventory files, PDFs)
2. Observe the crawler attempting to process non-XML files instead of skipping them

## 🖥 Environment Info
- Version of this software: 5.2.1-SNAPSHOT
- Operating System: Any

## 📚 Version of Software Used
5.2.1-SNAPSHOT

## 🩺 Test Data / Additional context
Affected class: `gov.nasa.pds.harvest.search.crawler.PDSProductCrawler`

## 🦄 Related requirements
N/A

---

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
Add a file extension check in `PDSProductCrawler` to skip non-XML files (CSV inventory files, PDFs, TXT files, etc.) before processing. Files that do not end with `.xml` are logged as SKIP and counted in `HarvestSolrStats.numFilesSkipped`, preventing unnecessary processing overhead.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [x] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: 50%

## ⚙️ Test Data and/or Report
Manually verified that running `harvest-solr` against a PDS4 data directory containing mixed file types (XML labels alongside CSV inventory files) correctly skips non-XML files and logs them as SKIP in the output.

## ♻️ Related Issues
Fixes #246

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: harvest-legacy/src/main/java/gov/nasa/pds/harvest/search/crawler/PDSProductCrawler.java [hunk 1]
```diff
@@ -260,6 +260,13 @@ protected boolean passesPreconditions(File product) {
     if (Constants.collections.contains(product)) {
       return false;
     }
+    // Skip non-XML files (e.g., CSV inventory files, PDFs, TXT files, etc.)
+    String fileName = product.getName().toLowerCase();
+    if (!fileName.endsWith(".xml")) {
```
Comment (nutjob4life) [inline harvest-legacy/src/main/java/gov/nasa/pds/harvest/search/crawler/PDSProductCrawler.java]: Would we ever have to worry about `.lblx` files?

**Review by nutjob4life — APPROVED** [review]
See the question about `.lblx` files, but otherwise this looks great and `mvn test` seems happy.

File: harvest-legacy/src/main/java/gov/nasa/pds/harvest/search/crawler/PDSProductCrawler.java (hunk 1 above)
Comment (jordanpadams) [reply harvest-legacy/src/main/java/gov/nasa/pds/harvest/search/crawler/PDSProductCrawler.java]: Good catch! Updated the check to also allow `.lblx` files alongside `.xml`. Case-insensitivity was already handled by the `.toLowerCase()` call on the filename, so `.XML`, `.LBLX`, etc. are all covered. See commit 61982619.
